data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java

```diff
@@ -220,6 +228,23 @@ public ApiResponse<String> makeRequest(
               authSetting,
               apiCallback);
     }
+
+    if (gZip) {
+      headersParams.put("Content-Encoding", "gzip");
+
+      try {
+
+        byte[] bytes = call.request().body().toString().getBytes(StandardCharsets.UTF_8);
+        ByteArrayOutputStream out = new ByteArrayOutputStream(bytes.length);
+        GZIPOutputStream gzip = new GZIPOutputStream(out);
```

Comment:
do we need to close this Stream in the end??

Reply:
Yes, Done